packages/circuit-ui/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright 2022, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  FieldsetHTMLAttributes,
+  InputHTMLAttributes,
+  Ref,
+  forwardRef,
+} from 'react';
+
+import styled from '../../styles/styled';
+import { typography } from '../../styles/style-mixins';
+import { uniqueId } from '../../util/id';
+import { Checkbox, CheckboxProps } from '../Checkbox/Checkbox';
+import {
+  FieldWrapper,
+  FieldLabelText,
+  FieldValidationHint,
+} from '../FieldAtoms';
+import { AccessibilityError } from '../../util/errors';
+
+export interface CheckboxGroupProps
+  extends Omit<FieldsetHTMLAttributes<HTMLFieldSetElement>, 'onChange'> {
+  /**
+   * A collection of available options. Each option must have at least a label and a value
+   * for the respective Checkbox.
+   */
+  options: (Omit<CheckboxProps, 'onChange'> & {
+    label: string;
+  })[];
+  /**
+   * The values of the Checkboxes that are checked by default.
+   */
+  value?: CheckboxProps['value'][];
+  /**
+   * A callback that is called when any of the checkboxes change their values.
+   * Passed on to the Checkboxes.
+   */
+  onChange: CheckboxProps['onChange'];
+  /**
+   * A description of the selector group.
+   */
+  label: string;
+  /**
+   * Label to indicate that the input group is optional. Only displayed when the
+   * `required` prop is falsy.
+   */
+  optionalLabel?: string;
+  /**
+   * The ref to the HTML DOM element.
+   */
+  ref?: Ref<HTMLFieldSetElement>;
+  /**
+   * Makes the input group required.
+   */
+  required?: InputHTMLAttributes<HTMLInputElement>['required'];
+  /**
+   * An information, warning or error message, displayed below the Checkboxes.
+   */
+  validationHint?: string;
+  /**
+   * Triggers error message below the Checkboxes.
+   */
+  invalid?: boolean;
+  /**
+   * Triggers warning message below the Checkboxes.
+   */
+  hasWarning?: boolean;
+  /**
+   * Triggers valid message below the Checkboxes.
+   */
+  showValid?: boolean;
+  /**
+   * Visually hide the label. This should only be used in rare cases and only
+   * if the purpose of the field can be inferred from other context.
+   */
+  hideLabel?: boolean;
+}
+
+const Legend = styled('legend')<Record<string, unknown>>(typography('two'));
```

Comment:
I think that's because `Legend` wraps around the `FieldLabelText` so `Legend` expects a `prop` named `children`. I can rewrite it to be something like so.
```
const Legend = styled('legend')<Record<'children', JSX.Element>>(
  typography('two'),
);
```
WDYT?

Reply:
Why does the generic type need to be provided at all? If TypeScript otherwise complains about the missing `theme` prop, you can use the `NoTheme` helper from `styles/styled`.